frontend/packages/metal3-plugin/src/components/dashboard/details-card.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+} from '@console/internal/components/dashboard/dashboard-card';
+import { DetailsBody, DetailItem } from '@console/internal/components/dashboard/details-card';
+import {
+  DashboardItemProps,
+  withDashboardResources,
+} from '@console/internal/components/dashboards-page/with-dashboard-resources';
+import { getName, getMachineNode } from '@console/shared';
+import { K8sResourceKind, MachineKind, NodeKind } from '@console/internal/module/k8s';
+import { getHostMachine, findNodeMaintenance } from '../../selectors';
+import { getHostStatus } from '../../utils/host-status';
+import MachineCell from '../machine-cell';
+import { BaremetalHostRole } from '../host-role';
+
+const DetailsCard: React.FC<DetailsCardProps> = ({ obj, machines, nodes, nodeMaintenances }) => {
+  const machine = getHostMachine(obj, machines);
+  const node = getMachineNode(machine, nodes);
+  const nodeName = getName(obj);
+  const nodeMaintenance = findNodeMaintenance(nodeMaintenances, nodeName);
+  const status = getHostStatus({ host: obj, machine, node, nodeMaintenance });
+  const machineCell = <MachineCell host={obj} status={status} />;
+  const hostRole = <BaremetalHostRole machine={machine} />;
+
+  return (
+    <DashboardCard>
+      <DashboardCardHeader>
+        <DashboardCardTitle>Details</DashboardCardTitle>
+      </DashboardCardHeader>
+      <DashboardCardBody>
+        <DetailsBody>
+          <DetailItem key="node-name" title="Node name" value={nodeName} isLoading={false} />
+          <DetailItem key="host-role" title="Role" value={hostRole} isLoading={false} />
+          <DetailItem key="machine-cell" title="Machine" value={machineCell} isLoading={false} />
```

Comment:
To align with relatively recent decision to link to a Node instead of Machine from the host, this should reference a Node using `NodeCell`

Reply:
:man_facepalming: